platform/wifi-emulator/platform_emulator.c

```diff
@@ -0,0 +1,667 @@
+{
+    return wifi_hal_getNeighboringWiFiStatus(radio_index, neighbor_ap_array, output_array_size);
+}
+
+int wifi_setQamPlus(void *priv)
+{
+    return 0;
+}
+
+int wifi_setApRetrylimit(void *priv)
+{
+    return 0;
+}
+
+
+INT wifi_getRadioChannelStats(INT radioIndex, wifi_channelStats_t *input_output_channelStats_array,
+    INT array_size)
+{
+    return RETURN_OK;
+}
+//--------------------------------------------------------------------------------------------------
+INT wifi_getApEnable(INT apIndex, BOOL *output_bool)
+{
+    return RETURN_OK;
+}
+
+//--------------------------------------------------------------------------------------------------
+INT wifi_setApMacAddressControlMode(INT apIndex, INT filterMode)
+{
+    return RETURN_OK;
+}
+
+
+//--------------------------------------------------------------------------------------------------
+INT wifi_getBssLoad(INT apIndex, BOOL *enabled)
+{
+    return RETURN_ERR;
+}
+
+//--------------------------------------------------------------------------------------------------
+INT wifi_setLayer2TrafficInspectionFiltering(INT apIndex, BOOL enabled)
+{
+    return RETURN_ERR;
+}
+//--------------------------------------------------------------------------------------------------
+INT wifi_setApIsolationEnable(INT apIndex, BOOL enable)
+{
+    return RETURN_ERR;
+}
+
+int platform_get_radio_caps(wifi_radio_index_t index)
+{ 
+    return 0;
+}
+
+INT wifi_getApDeviceRSSI(INT ap_index, CHAR *MAC, INT *output_RSSI)
+{
+    return 0;
+}
+
+static int get_sta_list_handler(struct nl_msg *msg, void *arg)
+{
+    struct nlattr *tb[NL80211_ATTR_MAX + 1];
+    struct genlmsghdr *gnlh = nlmsg_data(nlmsg_hdr(msg));
+    sta_list_t *sta_list = (sta_list_t *)arg;
+
+    if (nla_parse(tb, NL80211_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0),
+        NULL) < 0) {
+        wifi_hal_error_print("%s:%d Failed to parse sta data\n", __func__, __LINE__);
+        return NL_SKIP;
+    }
+
+    if (tb[NL80211_ATTR_MAC]) {
+        sta_list->macs = realloc(sta_list->macs, (sta_list->num + 1) * sizeof(mac_address_t));
+        if (sta_list->macs) {
+            memcpy(sta_list->macs[sta_list->num], nla_data(tb[NL80211_ATTR_MAC]), sizeof(mac_address_t));
+            sta_list->num++;
+        }
+    }
+
+    return NL_OK;
+}
+
+static int get_sta_list(wifi_interface_info_t *interface, sta_list_t *sta_list)
+{
+    int ret;
+    struct nl_msg *msg = NULL;
+
+    msg = nl80211_drv_cmd_msg(g_wifi_hal.nl80211_id, interface, NLM_F_DUMP, NL80211_CMD_GET_STATION);
+    if (msg == NULL) {
+        wifi_hal_error_print("%s:%d Failed to create NL command\n", __func__, __LINE__);
+        return -1;
+    }
+
+    ret = nl80211_send_and_recv(msg, get_sta_list_handler, sta_list, NULL, NULL);
+    if (ret < 0) {
+        wifi_hal_error_print("%s:%d Failed to execute NL command\n", __func__, __LINE__);
+        return -1;
+    }
+
+    return 0;
+}
+
+static int get_sta_stats_handler(struct nl_msg *msg, void *arg)
```

Comment:
There is a code match here.  Please add the credit here:
Adapted code from hostap, which is:
Copyright (c) 2002-2015, Jouni Malinen j@w1.fi et al. (see NOTICE),
Licensed under the BSD-3 License


Also, this file should have an RDK Apache header.